common/lib/xmodule/xmodule/modulestore/split_mongo/mongo_connection.py

```diff
@@ -287,25 +288,12 @@ def __init__(
         """
         Create & open the connection, authenticate, and provide pointers to the collections
         """
-        if kwargs.get('replicaSet') is None:
-            kwargs.pop('replicaSet', None)
-            mongo_class = pymongo.MongoClient
-        else:
-            mongo_class = pymongo.MongoReplicaSetClient
-        _client = mongo_class(
-            host=host,
-            port=port,
-            tz_aware=tz_aware,
-            **kwargs
-        )
-        self.database = MongoProxy(
-            pymongo.database.Database(_client, db),
-            wait_time=retry_wait_time
+        self.database = connect_to_mongodb(
+            db, collection, host,
+            port=port, tz_aware=tz_aware, user=user, password=password,
+            retry_wait_time=retry_wait_time, **kwargs
```

Comment:
Do we need the declared defaults here and in the lower level `connect_to_mongodb` definition?